docs/guide/advanced/accessibility.md

```diff
@@ -1,4 +1,158 @@
+
+## Meaningfulness
+
+Icons should represent concepts or actions in a universally understandable way. Avoid using abstract
+or ambiguous, or culture-specific symbols that might confuse some users.
+
+![For example: Use universally understandable symbols and don't base your choice of icon on puns.](../../images/a11y/meaningfulness.svg?raw=true)
+
+## Consistency
+
+Maintain consistency in icon design and usage across your interface to help users learn and
+understand their meanings more easily.
+
+![For example: Don’t use the same icon for multiple distinct purposes or meanings. Don’t use different icons for the same purpose or function.](../../images/a11y/consistency.svg?raw=true)
+
+## Text Alternatives
+
+You may have to provide text labels or alternative text descriptions for icons, especially for
+screen readers used by people with visual impairments.
+
+However: descriptions should only be provided to standalone icons that aren't purely decorative, as
+providing accessible names to non-functional elements only increases clutter when using screen
+readers.
+
+### On standalone icons
+
+Icons are usually very unlikely to stand on their own with no semantically meaningful wrapper
+element. In most cases they will be part of a badge, button (including icon buttons), navigation
+item or other interactive UI element.
+
+::: warning
+In case some of your icons stand alone, and they serve a non-decorative function, make sure to
+provide the appropriate accessible label for them.
+:::
+
+![In short: provide accessible label for semantic icons, but not for decorative icons.](../../images/a11y/alttext-standalone.svg?raw=true)
+
+In general try to avoid using functional icons with no interactivity, we recommend that:
+
+1) you either add a visible description next to them, or
+2) place them in badges, labels or on buttons, and at least add a tooltip to them.
+
+In any such case, it is preferred that the accessible name be provided for these interactive
+elements (badges, buttons, nav items etc.) only, _not_ the icons themselves.
+
+### On buttons
+
+Do not provide an accessible label to icons when used on a button, as this label will be read out by
+screen readers, leading to nonsensical text.
+
+![](../../images/a11y/alttext-buttons.svg?raw=true)
+
+### On icon buttons
+
+Icon buttons are buttons that do not contain any visible text apart from the icon itself (think of
+the close button of a dialog for example).
+
+As previously stated, you should provide your accessible label on the icon button itself, not the
+contained icon.
+
+![](../../images/a11y/alttext-iconbuttons.svg?raw=true)
+
+## A note on `aria-label`
+
+Although you could provide accessible labels to your elements via the `aria-label` attribute, we
+generally recommend against this for and instead suggest that
+you [use your chosen CSS framework's "visually hidden" utility whenever possible](https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/).
```

Comment:
Can you explain why we should be "against" this? It would be nice if we explained this more in the text, about why using "visually hidden" is better than using `aria-label`, we also could provide some sources for blog posts or documentation to prove our "recommendation".

Also maybe this reads a bit nicer
``` suggestion
generally recommend visually hiding textual descriptions. For example by [using utility CSS framework's "visually hidden"](https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/).
```

Reply:
That means something different though. Also, the link itself is the resource that explains why `.visually-hidden` > `[aria-label]`, I've rephrased the link text so that it's clearer.

```suggestion
generally recommend against this and instead suggest that you use your chosen CSS framework's "visually hidden" utility whenever possible. You can [read more about why `aria-label` might not be the best solution](https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/).
```

Reply:
I'm still not happy with the phraise "we generally recommend against this". That suggest that we are against the use of `aria-label`. And from what I found there nothing wrong with it.
I think you should approach it more positively. "If you are using a CSS framework like Tailwind, we recommend using visually hidden utilities whenever possible"

Reply:
The options from best to worst:

1. visible label
2. visually hidden label
3. aria-label

Reply:
> And from what I found there nothing wrong with it.

Well, there kind of is, that's what the article linked is about: https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/#the-aria-label-attribute-is-not-always-translated

But sure, we can rephrase it to be more lenient.

Reply:
They use the word "avoid" instead of "against", which is I think more right fit. Since using `aria-label` is not wrong. I  think we also should link this article to this paragraph, it gives a bit more context to this topic.

Reply:
No, in certain scenarios it might actually be wrong, since `aria-label` will generally be ignored on non-interactive elements. I don't believe linking to a certain paragraph is necessary, since the entirety of article is about why `aria-label` should be avoided if possible, there isn't a single reason, there are multiple.

Since this thread is on an outdated commit, I've made a suggestion here: https://github.com/lucide-icons/lucide/pull/2122/files#r1636386974

Reply:
Ah ok true, tnx for suggestion think it reads a bit nicer as well.